### PR TITLE
chore: bump to @favware/cliff-jumper v4.1.0 and fix changelog generation

### PR DESCRIPTION
**Pull request by favna**
**Please describe the changes this PR makes and why it should be merged:**

This updates @favware/cliff-jumper to v4.1.0. Not many breaking changes to process. It is only the added `--skip-changelog` to the `publish.dev.yml`. The `cliff.toml` change is a deduplication of the GitHub URL, but not strictly necessary.

I also cleaned up the `CHANGELOG.md` files, removing any duplicate entries. I then also found the fix for git-cliff no longer adding new duplicate entries by comparing the `cliff.toml`'s of this repo to the ones of the Sapphire Utilities monorepo and I noticed this repo set `topo_order` to `true` where Sapphire sets it to `false`. Matching Sapphire with this config I then ran `cliff-jumper` through `pnpm run release --skip-tag` on every relevant package and the changelog generated is perfectly clean.

<details>
<summary>Click here for a breakdown of packages (also posted on Discord)</summary>

- brokers :white_check_mark: (resolved version: 1.0.0)
- builders :white_check_mark: (resolved version: 1.9.0)
- collection :white_check_mark: (resolved version: 2.1.1)
- core :white_check_mark: (resolved version: 2.0.0)
- create-discord-bot :white_check_mark: (resolved version: 0.3.1)
- discord.js :white_check_mark: (resolved version: 14.16.0) - For some reason git-cliff thinks the previous version is `@discordjs/ws@1.1.1` which causes the header to be 
```md
# [14.16.0](https://github.com/discordjs/discord.js/compare/@discordjs/ws@1.1.1...14.16.0) - (2024-08-24)
```
from the template
```
/compare/{{ previous.version }}...{{ version }})
```
This is definitely a git-cliff problem as I can repro it with
```sh
git cliff --tag "14.16.0" --prepend ./CHANGELOG.md -u -c ./cliff.toml -r ../../ --include-path "packages/discord.js/*"
```
That said, the actual content of the changelog is correct. Only this one URL to a GitHub compare link is wrong.
- formatters :white_check_mark: (resolved version: 0.5.0)
- proxy :white_check_mark: (resolved version: 2.1.1) (note: no changelog changes because since 2.1.0 there have only `chore` and `build` commits)
- rest :white_check_mark: (resolved version: 2.4.0)
- util :white_check_mark: (resolved version: 1.1.1) (note: no changelog changes because since 1.1.0 there have only `chore` and `build` commits)
- voice :white_check_mark: (resolved version: 0.17.1) (note: no changelog changes because since 0.17.0 there have only `chore` and `build` commits)
- ws :white_check_mark: (resolved version: 2.0.0)

</details>

---

Lastly, I also added typescript as a dev-dep to `api-extractor-model` because otherwise `pnpm run build` fails because it cannot find the properly symlinked binary:

image

**Status and versioning classification:**
- This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Review by vladfrangu — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

File: packages/core/CHANGELOG.md [hunk 1]
```diff
@@ -125,154 +125,6 @@ All notable changes to this project will be documented in this file.
...
-- **core:** Include data for defer (#9284) ([9d69bba](https://github.com/discordjs/discord.js/commit/9d69bba47c73b756086992bc14e57c40fadb34d1))
-- **scripts:** Accessing tsComment ([d8d5f31](https://github.com/discordjs/discord.js/commit/d8d5f31d3927fd1de62f1fa3a1a6e454243ad87b))
-
-## Features
-
-- **core:** Http-only wrapper (#9281) ([11e682c](https://github.com/discordjs/discord.js/commit/11e682cfe388b9a3070388f73ebef3c27555c0dd))
-- **core:** Add `AbortSignal` support. (#9042) ([907eb1b](https://github.com/discordjs/discord.js/commit/907eb1b4708bdaf30f4e59f4016ef8a717f47a4c))
-- **website:** Render syntax and mdx on the server (#9086) ([ee5169e](https://github.com/discordjs/discord.js/commit/ee5169e0aadd7bbfcd752aae614ec0f69602b68b))
-
-# [@discordjs/core@0.4.0](https://github.com/discordjs/discord.js/compare/@discordjs/core@0.3.0...@discordjs/core@0.4.0) - (2023-03-12)
```
Comment (almeidx) [inline packages/core/CHANGELOG.md:248]: This version is missing in the CHANGELOG now

File: packages/discord.js/CHANGELOG.md [hunk 2]
```diff
@@ -688,4496 +688,6 @@ All notable changes to this project will be documented in this file.
...
-- **Guild:** Add `max_stage_video_channel_users` (#8422) ([34bc36a](https://github.com/discordjs/discord.js/commit/34bc36ac4b04ad033d0dcc3d2701fcf2e682743c))
-- **website:** Render syntax and mdx on the server (#9086) ([ee5169e](https://github.com/discordjs/discord.js/commit/ee5169e0aadd7bbfcd752aae614ec0f69602b68b))
-- Add GuildBasedTextChannelTypes (#9234) ([5f93dcc](https://github.com/discordjs/discord.js/commit/5f93dcce466286f0fdead8faf4131e98d1c9db55))
-
-## Refactor
-
-- Call `GuildBanManager#create()` directly (#9263) ([f340f3b](https://github.com/discordjs/discord.js/commit/f340f3b1fd719e8f7cf6fa28a41835bc16039fc7))
-- **FetchThreadsOptions:** Remove `active` (#9241) ([519e163](https://github.com/discordjs/discord.js/commit/519e163f8aa3b55420f86d2d60c2584b3a2eb327))
-
-# [14.8.0](https://github.com/discordjs/discord.js/compare/14.7.1...14.8.0) - (2023-03-12)
```
Comment (almeidx) [inline packages/discord.js/CHANGELOG.md:918]: This version is missing in the CHANGELOG

File: packages/discord.js/CHANGELOG.md [hunk 3]
```diff
@@ -688,4496 +688,6 @@ All notable changes to this project will be documented in this file.
...
-- Allow builders to set channel types in discord.js (#8990) ([7dec892](https://github.com/discordjs/discord.js/commit/7dec892218f7b470a5f8e78732a524a53da24d26))
-- Swap message reaction and emoji identifier types (#8969) ([ad49845](https://github.com/discordjs/discord.js/commit/ad4984526020f2baeefaeeebbded66c6848c4b85))
-- **widget:** Add missing `name` (#8978) ([898b5ac](https://github.com/discordjs/discord.js/commit/898b5ac416cbbb415b125bb27221d0901fdd180e))
-- Use StringSelectMenuOptionBuilder (#8949) ([bec51de](https://github.com/discordjs/discord.js/commit/bec51de1038c35c6edaaa13934781758fe1951de))
-- Fix actions type in automod (#8962) ([5915f39](https://github.com/discordjs/discord.js/commit/5915f39810b712c05a46fa21ab4e12b4cfa3c25a))
-- Subcommand group `options` is required (#8966) ([5dc5e90](https://github.com/discordjs/discord.js/commit/5dc5e902688fc563087cd5061dcb59dd68fd4eda))
-- Add generic to `ActionRowBuilder.from()` (#8414) ([153352a](https://github.com/discordjs/discord.js/commit/153352ad7a1ccb4a9461523cf2597d81df93b69c))
-
-# [14.7.1](https://github.com/discordjs/discord.js/compare/14.7.0...14.7.1) - (2022-12-01)
```
Comment (almeidx) [inline packages/discord.js/CHANGELOG.md:1012]: The version in the changelog doesn't have the URL:
https://github.com/discordjs/discord.js/blob/2106e914b2b650baa33d5bb3e43b72a82e0cb52a/packages/discord.js/CHANGELOG.md?plain=1#L418

File: packages/ws/CHANGELOG.md [hunk 4]
```diff
@@ -109,133 +109,6 @@ in non-node-like environments (CF Workers, etc.)
...
-## Bug Fixes
-
-- Fix external links (#9313) ([a7425c2](https://github.com/discordjs/discord.js/commit/a7425c29c4f23f1b31f4c6a463107ca9eb7fd7e2))
-
-## Refactor
-
-- Abstract identify throttling and correct max_concurrency handling (#9375) ([02dfaf1](https://github.com/discordjs/discord.js/commit/02dfaf1aa2c798315d0dd7f809cc469771b36ffc))
-- **WebSocketShard:** WaitForEvent and its error handling (#9282) ([dcf58d8](https://github.com/discordjs/discord.js/commit/dcf58d81401387a5e157b20829aa56638e106e9d))
-
-# [@discordjs/ws@0.8.0](https://github.com/discordjs/discord.js/compare/@discordjs/ws@0.7.0...@discordjs/ws@0.8.0) - (2023-04-01)
```
Comment (almeidx) [inline packages/ws/CHANGELOG.md:226]: This version is missing in the changelog

**Pull request merged**